[Docs]Add genAI use-case to docs (#395)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/genAI-use-case.md

# Vizro-AI use case

This guide provides instructions on incorporating a chart generated by VizroAI into your [vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.

> First step is to create a chart illustrating the population development of each continent over time. To gain deeper insights and access the underlying code responsible for generating the chart, include explain=True in the plot() method. Let's execute the provided code and examine the outcome.

Reviewer: ```suggestion
First step is to create a chart illustrating the population development of each continent over time. To gain deeper insights and access the underlying code responsible for generating the chart, include `explain=True` in the `plot()` method. Let's execute the provided code and examine the outcome.
```

!!! example "Vizro-AI chart"
    === "Code for the cell"
        ```py
        import vizro_ai
        from vizro_ai import VizroAI
        import plotly.express as px
        from dotenv import load_dotenv

        load_dotenv()

        df = px.data.gapminder()
        vizro_ai = VizroAI(model_name="gpt-4-0613")

        vizro_ai.plot(df,
                      """Plot a bubble chart to show the changes in life expectancy
                       and GDP per capita for each country over time.
                       Color the bubbles by continent.
                       Add animation on yearly basis, and do not use facets.
                       Put the legend on top""")
        ```
    === "Result"
        [![VizroAIChart]][VizroAIChart]

    [VizroAIChart]: ../../assets/user_guides/vizro-ai-chart.gif

> Once you are satisfied with the chart, we can proceed with the next step, and that is incorporating the chart into [vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.

Reviewer 2: ```suggestion
2. Insert the resulting chart into a dashboard. Once you are satisfied with the chart, you incorporate the chart into a [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.
```

> In the explanation section, you'll find a code snippet used to generate the chart.

Reviewer: ```suggestion
In the explanation section from the result of the example above, you'll find a code snippet used to generate the chart.
```

> In the explanation section, you'll find a code snippet used to generate the chart.

Reviewer 2: ```suggestion
In the explanation section, you'll find the code snippet used to generate the chart.
```

```py
@capture('graph')
def custom_chart(data_frame=pd.DataFrame()):
    # Create a bubble chart
    fig = px.scatter(data_frame, x='gdpPercap', y='lifeExp', animation_frame='year', animation_group='country',
                     size='pop', color='continent', hover_name='country', log_x=True, size_max=60,
                     range_y=[20,90],
                     labels={'gdpPercap':'GDP per Capita', 'lifeExp':'Life Expectancy', 'year':'Year', 'continent':'Continent', 'pop':'Population'},
                     title='Life Expectancy v. Per Capita GDP, 1952-2007',
                     height=400)
    # Put the legend on top
    fig.update_layout(legend=dict(y=1, yanchor='top', x=0.5, xanchor='center'))
    # Return the plot
    return fig
```

> Simply copy and paste the custom chart code as it can be used directly in the [vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.

Reviewer: Should we mention how the `custom_chart` from the example has to be places as vm.Graph figure attribute, and then also link vm.Graph model?

> Simply copy and paste the custom chart code as it can be used directly in the [vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.

Reviewer 2: ```suggestion
Simply copy and paste the custom chart code as it can be used directly in the [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.
```

                             height=400)
            # Put the legend on top
            fig.update_layout(legend=dict(y=1, yanchor='top', x=0.5, xanchor='center'))
            # Return the plot
            return fig

        df = px.data.gapminder()

        page = vm.Page(
            title = 'Demographics',
            components = [
                vm.Graph(id='bubble chart', figure=custom_chart(df)),
                vm.Graph(id='histogram', figure = px.box(df,
                                                         x='continent',
                                                         y='lifeExp',
                                                         color='continent',
                                                         title='Life Expectancy per Continent'))],
            controls = [
                vm.Filter(column='country'),
                vm.Filter(column='continent')])

        Vizro().build(vm.Dashboard(pages=[page])).run(port=8000)
        ```
    === "Result"
        [![VizroDashboard]][VizroDashboard]

    [VizroDashboard]: ../../assets/user_guides/chart_into_dashboard_large.gif

> With Vizro-AI you can create visually captivating charts and plug it in your [vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard in seconds!

Reviewer 2: ```suggestion
With Vizro-AI you can create visually captivating charts and plug them into your [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard in seconds!
```

> # How to add Vizro-AI created charts into a Vizro dashboard

Reviewer 3: Would it be possible to rename the file to make it more obvious what this page explains something like "add-generated-chart-usecase.md" perhaps?

> This guide provides instructions on incorporating a chart generated by Vizro-AI into your [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.

Reviewer 3: ```suggestion
This guide explains how to add a chart generated by Vizro-AI to your [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.
```

1. Create a chart with Vizro-AI that you would like to visualize in a dashboard.

>     In this example, we aim to create a chart that illustrates the population development of each continent over time. To gain deeper insights and access the underlying code responsible for generating the chart, include `explain=True` in the `plot()` method. Let's execute the provided code and examine the outcome.

Reviewer 3: I think you should include `explain=True` it in the code snippet below since you mention it here.

Author: Good catch. 🙌

    !!! example "Vizro-AI chart"
        === "Code for the cell"
            ```py
            import vizro_ai
            from vizro_ai import VizroAI
            import plotly.express as px
            from dotenv import load_dotenv

            load_dotenv()

            df = px.data.gapminder()
            vizro_ai = VizroAI(model_name="gpt-4-0613")

            vizro_ai.plot(df,
                          """Plot a bubble chart to show the changes in life expectancy
                           and GDP per capita for each country over time.
                           Color the bubbles by continent.
                           Add animation on yearly basis, and do not use facets.
                           Put the legend on top""")
            ```
        === "Result"
            [![VizroAIChart]][VizroAIChart]

        [VizroAIChart]: ../../assets/user_guides/vizro-ai-chart.gif

2. Insert the resulting chart into a dashboard.

>     Once you are satisfied with the chart, you incorporate the chart into a [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard. In the explanation section from the result of the example above, you'll find the code snippet used to generate the chart.

Reviewer 3: Is there an explanation section if you don't pass `explain=True` to the `plot()` method?

Reviewer 3: ```suggestion
    Once you are satisfied with the chart, you can add it to a [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard. In the explanation section from the result of the example above, you'll find the code snippet used to generate the chart.
```

                         size='pop', color='continent', hover_name='country', log_x=True, size_max=60,
                         range_y=[20,90],
                         labels={'gdpPercap':'GDP per Capita', 'lifeExp':'Life Expectancy', 'year':'Year', 'continent':'Continent', 'pop':'Population'},
                         title='Life Expectancy v. Per Capita GDP, 1952-2007',
                         height=400)
        # Put the legend on top
        fig.update_layout(legend=dict(y=1, yanchor='top', x=0.5, xanchor='center'))
        # Return the plot
        return fig
    ```

The provided `custom_chart` function is an example of the [custom chart](https://vizro.readthedocs.io/en/stable/pages/user-guides/custom-charts), and it returns a `go.Figure()` object.
This object must then be passed to the `figure` argument of the Vizro [Graph](https://vizro.readthedocs.io/en/stable/pages/user-guides/graph) model.

!!! example "Vizro-AI chart within vizro dashboard"
    === "Code for the cell"
        ```py
        from vizro import Vizro
        import vizro.models as vm
        from vizro.models.types import capture
        import pandas as pd
        import plotly.express as px

>         @capture('graph')

Reviewer 3: I really like this approach. I put a similar very brief illustration in the core docs to show how it could be done if you just grab the call to Plotly and include that -- I'll add you to my PR to see what you think. Is it worth noting here that you could do that if you understand the code and want to minimize what you copy across?
